src-java/nbworker-topology/nbworker-storm-topology/src/main/java/org/openkilda/wfm/topology/nbworker/validators/PathValidator.java

```diff
@@ -0,0 +1,513 @@
+                        getValidations(pathValidationData)))
+                .flatMap(Set::stream)
+                .collect(Collectors.toSet());
+
+        return PathValidationResult.builder()
+                .isValid(result.isEmpty())
+                .errors(new LinkedList<>(result))
+                .build();
+    }
+
+    private Set<String> executeValidations(InputData inputData,
+                                           RepositoryData repositoryData,
+                                           List<BiFunction<InputData, RepositoryData, Set<String>>> validations) {
+
+        return validations.stream()
+                .map(f -> f.apply(inputData, repositoryData))
+                .flatMap(Set::stream)
+                .collect(Collectors.toSet());
+    }
+
+    private List<BiFunction<InputData, RepositoryData, Set<String>>> getValidations(
+            PathValidationData pathValidationData) {
+        List<BiFunction<InputData, RepositoryData, Set<String>>> validationFunctions = new LinkedList<>();
+
+        validationFunctions.add(this::validateForwardAndReverseLinks);
+
+        if (isLatencyValidationRequired(pathValidationData)) {
+            validationFunctions.add(this::validateLatency);
+        }
+
+        if (isLatencyTier2ValidationRequired(pathValidationData)) {
+            validationFunctions.add(this::validateLatencyTier2);
+        }
+
+        if (isBandwidthValidationRequired(pathValidationData)) {
+            validationFunctions.add(this::validateBandwidth);
+        }
+
+        if (isDiverseWithFlowValidationRequired(pathValidationData)) {
+            validationFunctions.add(this::validateDiverseWithFlow);
+        }
+
+        if (isEncapsulationTypeValidationRequired(pathValidationData)) {
+            validationFunctions.add(this::validateEncapsulationType);
+        }
+
+        return validationFunctions;
+    }
+
+    private boolean isEncapsulationTypeValidationRequired(PathValidationData pathValidationData) {
+        return pathValidationData.getFlowEncapsulationType() != null;
+    }
+
+    private boolean isDiverseWithFlowValidationRequired(PathValidationData pathValidationData) {
+        return StringUtils.isNotBlank(pathValidationData.getDiverseWithFlow());
+    }
+
+    private boolean isBandwidthValidationRequired(PathValidationData pathValidationData) {
+        return pathValidationData.getBandwidth() != null
+                && pathValidationData.getBandwidth() != 0
+                && (pathValidationData.getPathComputationStrategy() == null
+                || pathValidationData.getPathComputationStrategy() == COST_AND_AVAILABLE_BANDWIDTH);
+    }
+
+    private boolean isLatencyTier2ValidationRequired(PathValidationData pathValidationData) {
+        return pathValidationData.getLatencyTier2() != null
+                && !pathValidationData.getLatencyTier2().isZero()
+                && (pathValidationData.getPathComputationStrategy() == null
```

Comment:
I made it so that we can execute all validations if we don't know what strategy to use. With defaulting it's trickier to interpret results because it is not clear which validations have been executed, or why not: there is no strategy in the response. 
Do you want to add this strategy defaulting requirement to this task?

Reply:
But our PCE can't work without strategy. If you think that it's trickier to interpret results with defaulting we can mark `strategy` param as mandatory. 

But mostly I'm looking at current PCE. For example. An operator want to check some path and send a requrst without strategy, but wit latency. PCE says that latany is too small and the path is bad. But when the operator send same request to flow create API - flow will be created because default strategy is `cost` and it doesn't care about latency

Reply:
I think you should remove line `pathValidationData.getPathComputationStrategy() == null`